rpm/gen-fixtures-delta.sh

```diff
@@ -0,0 +1,160 @@
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    last_part=${parts[-1]}
+    pattern="${2}.rpm"
+
+    release=${last_part%.$pattern}
+    echo "${release}"
+}
+
+# Given a path to an RPM package, echo the package arch.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → noarch
+get_rpm_arch() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr . "\n"))
+    echo "${parts[-2]}"
+}
+
+#------------------------------------------------------------------------------
+# Business Logic
+#------------------------------------------------------------------------------
+
+# Fetch output_dir from user.
+if [ "$#" -lt 2 ]; then
+    echo 1>&2 'Error: Too few arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+elif [ "$#" -gt 3 ]; then
+    echo 1>&2 'Error: Too many arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+else
+    output_dir="$(realpath --canonicalize-missing "${1}")"
+    assets_dir="$(realpath "${2}")"
+    package_name="${3}"
+fi
+
+# Copy all RPMs from assets_dir to output_dir
+# Create directory 'drpms' for DRPMS
+mkdir -vp "${output_dir}"/drpms
+cp -t "${output_dir}" "${assets_dir}/"*.rpm
+
+# Filter and sort all required RPMs.
+rpms="${output_dir}"/"${package_name}"*.rpm
+IFS=$'\n' rpms=($(sort <<<"${rpms[*]}"))
+
+found_rpms=()
+for (( i=0; i<${#rpms[@]}; i++ ));
+do
+    current_package_name="$(get_rpm_name "${rpms[i]}")"
+    if [ "${current_package_name}" == "${package_name}" ]; then
+        found_rpms+=("${rpms[i]}")
+    fi
+done
+
+readonly num_needed=2
+rpms_len=${#found_rpms[@]}
+if [ ${rpms_len}  -lt ${num_needed} ]; then
+	echo 1>&2 'Error: Not found enought RPM packages to DRPMS. At least 2 RPMS is required'
+	exit 1
+fi
+
+# Make DRPMS from RPMS
+# DRPMS is generated from all neighbouring RPMS in filtered RPMS list
+for (( i=0; i<${rpms_len}-1; i++ ));
+do
+	old_package=${found_rpms[i]}
+	new_package=${found_rpms[i+1]}
+
+    old_name="$(get_rpm_name "${old_package}")"
+    new_name="$(get_rpm_name "${old_package}")"
+
+    if [ "${old_name}" != "${new_name}" ]; then
+        echo 1>&2 'Error: Packages need has same name'
+        exit
+    fi
+
+	old_arch=$(get_rpm_arch "${old_package}")
+	new_arch=$(get_rpm_arch "${new_package}")
+
+	if [ "${old_arch}" != "${new_arch}" ]; then
+		echo 1>&2 'Error: Old and new packages need has same arch'
+		exit
+	fi
+
+	old_version="$(get_rpm_version "${old_package}")"
+	new_version="$(get_rpm_version "${new_package}" )"
+
+	old_release="$(get_rpm_release "${old_package}" "${old_arch}")"
+	new_release="$(get_rpm_release "${new_package}" "${new_arch}")"
+
+	delta_package="${new_name}-${old_version}-${old_release}_${new_version}-${new_release}.${new_arch}.drpm"
+
+	makedeltarpm ${old_package} ${new_package} "${output_dir}"/drpms/"${delta_package}"
+done
+
+# Generate DRPMS fixtures from RPMS and DRPMS
+createrepo --checksum sha256 --delta ${output_dir}
```

Comment:
Quote all variable expansions to prevent word splitting issues. (Try cloning all of your repositories into a directory named `my code`, sometime, and see how many things break. It's astounding.)